x/auth/msgs_test.go

```diff
@@ -0,0 +1,47 @@
+package auth
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	crypto "github.com/tendermint/go-crypto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func TestNewMsgChangeKey(t *testing.T) {}
+
+func TestMsgChangeKeyType(t *testing.T) {
+	addr1 := sdk.Address([]byte("input"))
+	newPubKey := crypto.GenPrivKeyEd25519().PubKey()
+
+	var msg = MsgChangeKey{
+		Address:   addr1,
+		NewPubKey: newPubKey,
+	}
+
+	assert.Equal(t, msg.Type(), "auth")
+}
+
+func TestMsgChangeKeyValidation(t *testing.T) {
+
+	addr1 := sdk.Address([]byte("input"))
+
+	// emptyPubKey := crypto.PubKeyEd25519{}
```

Comment:
Why did we comment this testcase?

Reply:
An Empty pubkey still has prefix bytes.

Reply:
That's fine - I mean we should check `.IsValid()` (which may need to be added to tmlibs).